pkg/virt-handler/cmd-client/client.go

```diff
@@ -300,10 +300,20 @@ func (c *VirtLauncherClient) GetDomain() (*api.Domain, bool, error) {
 	request := &cmdv1.EmptyRequest{}
 	ctx, cancel := context.WithTimeout(context.Background(), shortTimeout)
 	defer cancel()
+
 	response, err := c.v1client.GetDomain(ctx, request)
+	var responseError error
+	if err != nil {
+		responseError = fmt.Errorf("unknown error encountered sending command %q: %s", "GetDomain", err.Error())
+		if IsDisconnected(err) {
+			responseError = err
+		}
+	} else if response != nil && response.Response.Success != true {
+		responseError = fmt.Errorf("server error. command %q failed: %q", "GetDomain", response.Response.Message)
+	}
 
-	if err = handleError(err, "GetDomain", response.Response); err != nil {
-		return domain, exists, err
+	if responseError != nil {
+		return domain, exists, responseError
```

Comment:
I somewhat dislike that we have the same code at 3 places now (here, `GetDomainStats` and `handleError`
What about:
```
	domainResponse, err := c.v1client.GetDomain(ctx, request)
	var response *cmdv1.Response
	if domainResponse != nil {
		response = domainResponse.Response
	}
	if err = handleError(err, "GetDomain", response); err != nil {
		return domain, exists, err
	}
```
and check if response is nil in `handleError`? Same in `GetDomainStats` then.